client/src/components/CardsContainer.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import VideoCard from "./VideoCard";
+
+// how to change a videoData to a useState variable
+// send that down to VideoCard because it has the button to delete
+/*
+we have 2 choices to delete the particular VideoCard:
+1. pass the full data down to delete a small part from the full data
+2. instead of passing the full data down, we get the unique number of the card to delete it from the full data
+
+We turn the videoData into a useState to delete the video with a specific data.
+How do you delete a record from a useState array (answer might be spread operator, perhaps .map?)
+*/
+
+function CardsContainer( { videoData, setVideoData }) {
+
+function handleDelete(id) {
```

Comment:
what you can do as an additional measure of security is check if the `id` exists inside this function. 
Something along these lines:
  ```
if (!id) {
    // Handle the case where id is not provided or is invalid
    // You can return, throw an error, or take any appropriate action here.
    return;
  }
  
  setVideoData(
    videoData.filter((video) => {
      return video.id !== id;
    })
  );
```
  
The code is great as it is, but you can consider it for the future use.

Reply:
Thank you!